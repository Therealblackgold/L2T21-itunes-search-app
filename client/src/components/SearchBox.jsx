// SearchBox component renders a search input to get items using the entity api query.
const SearchBox = ({ termValue, setTermValue }) => {
  return (
    <div className="col col-sm-4">
      <input
        type="text"
        className="form-control"
        placeholder="Search"
        value={termValue}
        onChange={(event) => setTermValue(event.target.value)}
      />
    </div>
  );
};

export default SearchBox;
