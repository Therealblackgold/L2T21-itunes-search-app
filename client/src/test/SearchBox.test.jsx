import { render, screen, fireEvent } from "@testing-library/react";
import SearchBox from "../components/SearchBox";
import renderer from "react-test-renderer";

const mockedSetTermValue = jest.fn();

describe("ItemList Component", () => {
  test("Should be able to type into input", () => {
    render(<SearchBox setTermValue={mockedSetTermValue} />);
    const inputElement = screen.getByPlaceholderText(/search/i);
    fireEvent.change(inputElement, { target: { value: "Jane Doe" } });
    expect(inputElement.value).toBe("Jane Doe");
  });

  test("Component renders properly", () => {
    const component = renderer.create(
      <SearchBox setTermValue={mockedSetTermValue} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
