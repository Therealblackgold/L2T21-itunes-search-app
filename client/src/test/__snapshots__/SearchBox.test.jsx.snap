// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList Component Component renders properly 1`] = `
<div
  className="col col-sm-4"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search"
    type="text"
  />
</div>
`;
